tests/testthat/helper.R

```diff
@@ -1,16 +1,37 @@
 Sys.setlocale("LC_COLLATE", "C") ## What CRAN does
 set.seed(666)
 
-## Our "test package" common harness code
-## `try` added because of some devtools::document weirdness
-try({
-    crunch_test_path <- system.file("crunch-test.R", package="crunch")
-    if (crunch_test_path == "") {
-        # hack for loadall
-        crunch_test_path <- "../../inst/crunch-test.R"
+# find a file starting with the current directory and looking (recursively) in parent folders limited to levels_up
+find_file <- function (file_name, levels_up = 3) {
+    for (i in 0:levels_up) {
+        if (i == 0) {
+            pth <- "."
+            
+        } else {
+            pth <- file.path(".", paste0(rep("..", i), collapse = "/"))
+        }
+        fls <- list.files(pth, recursive = TRUE)
+        fl <- fls[grepl(file_name, fls)]
+        if (length(fl) == 1) return(file.path(pth,fl))
+        if (length(fl)  > 1) {
+            warning("More than one file named ", file_name, " using the first one.")
```

Comment:
On what file system would it be possible to have two files with the same name?

Reply:
Because the search is recursive, the same file could be found in more than one directory. This is probably overly-defensive since we are the only ones who can't always just reference the installed file.